py/dml/structure.py

```diff
@@ -2199,6 +2200,47 @@ def explode_offsets(reg):
             if a[1] > b[0]:
                 report(EREGOL(a[2], b[2], a[3], b[3]))
 
+def check_register_fields(reg):
+    assert dml.globals.dml_version != (1, 2)
+    bitsize = param_int(reg, 'bitsize')
+    def explode_ranges(field):
+        param = field.get_component('lsb')
+        assert param.objtype == 'parameter'
+        ranges = []
+        for field_indices in itertools.product(
+                *([mkIntegerLiteral(field.site, i) for i in range(dimsize)]
+                  for dimsize in field.dimsizes[reg.dimensions:])):
+            reg_indices = tuple(StaticIndex(field.site, var)
+                                for var in reg.idxvars())
+            indices = reg_indices + field_indices
+            lsb_expr = param_expr(field, 'lsb', indices)
+            msb_expr = param_expr(field, 'msb', indices)
+            for expr in (lsb_expr, msb_expr):
+                if isinstance(expr, NonValue):
+                    raise expr.exc()
```

Comment:
I think this line lacks a test, e.g. a field `@ [undefined:undefined]`. Bonus points if you can trigger DCE (unmappable bank, unmapped register, or register that overrides get/set/read/write).

Reply:
I'll add a test, and try to trigger DCE. I'm not sure the latter is possible, though.

Reply:
Tested to see if DCE avoids the error by testing it with an unmappable bank with unmapped register that overrides read and write. No sell.